pkg/gitreceive/run.go

```diff
@@ -48,9 +50,16 @@ func Run(conf *Config, fs sys.FS, env sys.Env, storageDriver storagedriver.Stora
 
 		log.Debug("read [%s,%s,%s]", oldRev, newRev, refName)
 
-		if err := receive(conf, builderKey, newRev); err != nil {
+		client, err := controller.New()
```

Comment:
can we create the client outside of this for loop and pass that to the build function instead of creating again in the build function?